packages/account/src/account.ts

```diff
@@ -286,6 +292,13 @@ export class Account extends AbstractAccount {
       fundingAttempts += 1;
     }
 
+    if (bn(request.inputs.length).gt(maxInputs)) {
```

Comment:
Rather than checking it on the `Account`, I suggest you check it on the `Provider` in `Provider.sendTransaction` before the transaction is encoded [here](https://github.com/FuelLabs/fuels-ts/blob/ece23765948da6b35419cfde8f70a1999b704183/packages/account/src/providers/provider.ts#L710). You can create a helper function:
```ts
validateTransaction(tx: TransactionRequest, consensusParameters: ConsensusParameters) {
  if(bn(request.inputs.length).gt(consensusParameters.txParameters.maxInputs)) 
    { throw new FuelError(...); }
}
```

That is the place where the transaction is encoded and we are certain that it won't change.

Reply:
Makes sense. I will do this.

Reply:
@mvares Sure!